Update README.md (#165)
cognitedata/cdf-project-templates: README.md

# Official Cognite Data Fusion project templates
# Official Cognite Data Fusion Project configuration templates

The CDF Toolkit is a command-line interface (`cdf-tk`) used for configuring and administrating Cognite Data
Fusion (CDF) projects.
It supports three different modes of operation:
> Configure Cognite Data Fusion quickly, consistently, traceably, and repeatably.
 
> [!WARNING]
> The templates and the `cdf-tk` tool are currently in **ALPHA**. The scope in alpha is on Asset
Performance Management focused on Infield (Digital Operator Rounds). The templates and tooling
will be continously improved throughout moving towards beta and general availability.

> 1. As an **interactive command-line tool** used alongside the Cognite Data Fusion web application to retrieve and

Reviewer: Why don't you want the description of the three different use cases here?

Author: For brevity. I think of this as the stepwise guide, TLDR version

> This are the steps to get started with Cognite Data Fusion configuration:

Reviewer: ```suggestion
These are the steps to get started with Cognite Data Fusion configuration:
```

> This are the steps to get started with Cognite Data Fusion configuration:

Reviewer: When you have a list of steps first, and then you detail out those below, you will always loose a certain percentage who will not read on, but still be confused. Thus, all step descriptions should be actionable for those who do not want to read on.

> - Install the Cognite Data Fusion Toolkit in your repository.

Reviewer: ```suggestion
- Install the Cognite Data Fusion Toolkit on your machine using `pip install cognite-toolkit`.
```

> - Use run the `init` command to get the modular resource packages to install in your Cognite Data Fusion projects

Reviewer: ```suggestion
- Run the `cdf-tk init <new_dir>` command replacing <new_dir> with where you want the template configurations to install in your Cognite Data Fusion projects
```

- Configure the included (and add your own) modules by editing the yaml configuration files to fit your needs

> - Build, Verify and Deploy the configuration using the Toolkit

Reviewer: ```suggestion
- Build, Verify and Deploy the configuration using the `cdf-tk` tool. Use `cdf-tk --help` to learn more about the supported commands.
```

> - Optional: set up automated deployment using GitHub Actions

Reviewer: ```suggestion
- Optional: set up automated deployment using GitHub Actions or your CI/CD system of choice.
```

cdf-tk --help
```

Follow instructions on [Developer Toolkit Quickstart](https://developer.cognite.com/sdks/toolkit/quickstart).

## Step 2: Configure modules

A **module** is a bundle of Cognite Data Fusion resources that are coupled together as logical units, for example data pipelines or the configuration of an application. For convenience, related modules are combined in _packages_. See [Developer Toolkit Templates](https://developer.cognite.com/sdks/toolkit/templates) for a more extensive description. 

To install the modules, run this command with <my_project_dir> of your own choosing:

```sh
cdf-tk init <my_project_dir> 
```

### Select which modules to deploy

Edit the `<my_project_dir>/local.yaml` file. You can add multiple environments (Cognite Data Fusion Projects[^1]) like this:

```yaml
# environment name
dev:
  project: <customer>-dev # the cdf project name
  type: dev
  deploy: # packages or modules to deploy to this environment
    - cdf_demo_infield 
    - cdf_apm_simple
```

The environment (dev in the example) is used by the cli tool as `--env <environment>` to determine which configurations to deploy.

### Configure each module as needed

> Each module follow a similar pattern:

Reviewer: ```suggestion
Each module follows a similar pattern:
```

```
./modules|common|examples|local_modules/<moduleA>/
                                                |- data_models/
                                                |- auth/
                                                |- transformations/
                                                |- raw/
                                                |- time_series/
                                                |- files/
                                                default.config.yaml 
                                                config.yaml
```

> Create a copy of default.config.yaml to config.yaml and edit it according to the customer setup. In general, this is the only file that should be edited in the modules. The build command will replace placeholder values in the templates with these values.     

Reviewer: I believe we supply a default config.yaml, right? So this will confuse some people :)

> [!WARNING]
>
> `default.config.yaml` files and the contents in the resource folders should not be edited as they might be overwritten by toolkit updates.
> The correct way to configure a module is to make a copy of `default.config.yaml` as `config.yaml` and make changes there. These files should be added to version control.

#### 2.3 Create your own custom modules

Following the same pattern as above, custom modules can be added to `<my_project_dir>/local_modules/`. If you add these modules to your local.yaml, they will get included in the build. See advanced usage of templates in the [Developer Toolkit Advanced usage of templates](https://developer.cognite.com/sdks/toolkit/advanced).

## Step 3: Build, verify and deploy the configuration

Run the `build` command with the `cdf-tk` CLI:

```sh

> cdf-tk build --env <my_env> --build-dir <my_build_dir> 

Reviewer: You should include the source_dir as an argument maybe? Or comment that you need to run cdf-tk from the project dir. Maybe drop --build-dir as it will default be ./build?
